### PR TITLE
Add search tool in editor

### LINKED ISSUE #5523: Navigate using search bar in editors map
## Description

Having the search button (available in the "Map Viewer")  in the editor

## AC

- [ ] Same design as the search bar in the general map
- [ ] Search by stations, lines and signals
- [ ] When the result of the search is found, the map will zoom on the zone of the searched object (as in the general map)
- [ ] When the searched object is in a layer which in not activated, this layer will be automatically activated

### PR DESCRIPTION
**Pull request by Akctarus**
closes #5523  

- icon replacement with Go ones

### CODE REVIEW

**Review by tykayn — APPROVED** [review]

**Review by Yohh — APPROVED** [review]
nice! LGTM :muscle:

File: front/src/applications/editor/Editor.scss [hunk 1]
```diff
@@ -142,6 +142,10 @@
     & > .map {
       flex-grow: 1;
 
+      .map-modal {
+        max-width: 100%;
```
Comment (nicolaswurtz) [inline front/src/applications/editor/Editor.scss]: image

It's a bit too much no ? :laughing: 
Try to define some dynamic value, you could find inspiration on https://blog.logrocket.com/flexible-layouts-without-media-queries/ by example

File: front/src/applications/editor/nav.tsx [hunk 2]
```diff
@@ -15,6 +14,7 @@ import { selectLayers } from 'reducers/editor';
 import { Shortcut } from 'utils/hooks/useKeyboardShortcuts';
 import { ModalContextType } from 'common/BootstrapSNCF/ModalSNCF/ModalProvider';
 import InfraSelectorModal from 'common/InfraSelector/InfraSelectorModal';
+import { GoSearch, GoStack, GoZoomIn, GoZoomOut } from 'react-icons/go';
```
Comment (nicolaswurtz) [inline front/src/applications/editor/nav.tsx:17]: GG!

**Review by nicolaswurtz — CHANGES_REQUESTED** [review]

**Review by nicolaswurtz — APPROVED** [review]

File: front/src/applications/editor/Editor.scss (hunk 1 above)
Comment (Akctarus) [reply front/src/applications/editor/Editor.scss]: Fixed !

**Pull request merged**